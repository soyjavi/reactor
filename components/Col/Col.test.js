import React from 'react';
import renderer from 'react-test-renderer';

import { Col } from '..';

describe('<Col>', () => {
  it('renders', () => {
    const tree = renderer.create(<Col />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {alignContent}', () => {
    const tree = renderer.create(<Col alignContent="center" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {alignItems}', () => {
    const tree = renderer.create(<Col alignItems="center" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {justifyContent}', () => {
    const tree = renderer.create(<Col justifyContent="center" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('hook:styler', () => {
    const snapshot = renderer.create(<Col marginBottom="M" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
