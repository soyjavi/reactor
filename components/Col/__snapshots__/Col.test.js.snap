// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col> hook:styler 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignContent": "flex-start",
      },
      Object {
        "alignItems": "flex-start",
      },
      Object {
        "justifyContent": "flex-start",
      },
      Object {
        "marginBottom": 16,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Col> renders 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignContent": "flex-start",
      },
      Object {
        "alignItems": "flex-start",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Col> when {alignContent} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignContent": "center",
      },
      Object {
        "alignItems": "flex-start",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Col> when {alignItems} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignContent": "flex-start",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
/>
`;

exports[`<Col> when {justifyContent} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      Object {
        "alignContent": "flex-start",
      },
      Object {
        "alignItems": "flex-start",
      },
      Object {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
